src/Mutator/Util/Result/MutatorResult.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator\Util\Result;
+
+use PhpParser\Node;
+
+/**
+ * @internal
+ */
+interface MutatorResult
```

Comment:
My 5 cents:
What about to call an Interface like `MutatedNode` and create same class. Because `MutatorResult` it's smth abstract and doesn't reflect what is it and what for.

Reply:
I also agree that `MutatorResult` is too common name, but `MutatedNode` is also confusing when `\Generator` is used, even when `Node[]` is used (because it's not singular but plural, while `MutatedNode` is strictly singular)

What if we rename it in a way as it is a strategy, not the result of something:

```php
interface NodeReplacer {
    public function replaceWithNodes(): \Generator; // or getReplacedNodes(): \Generator
}
```

#### For a single Node 

```php
class SingleNodeReplacer implements NodeReplacer {
    public function __construct(Node $node) { ... }

    public function replaceWithNodes(): \Generator {
        yield $this->node;
    }
}
```

#### for array of Nodes

```php
class ArrayNodeReplacer implements NodeReplacer {
    public function __construct(Node[] $node) { ... }

    public function replaceWithNodes(): \Generator {
        yield $this->nodes;
    }
}
```
#### for Generator

```php
class GeneratorNodeReplacer implements NodeReplacer {
    public function __construct(\Generator $nodes) { ... }

    public function replaceWithNodes(): \Generator {
        return $this->nodes;
    }
}
```

client coude would be

```php
...
$nodeReplacer = $mutator->mutate($node);

foreach ($nodeReplacer->replaceWithNodes()  as $node) { ... }

// or

foreach ($nodeReplacer->getReplacedNodes()  as $node) { ... }

Reply:
Probably, the previous comment is too much.

Another ideas:

* `MutatedAst` as an interface name (which implies singular/plural)

Or, if we continue PHP-Parser wording (see [`NodeTraverser`](https://github.com/nikic/PHP-Parser/blob/4fb48c0e18ed69bb60ec27137e0875b0e23637e9/lib/PhpParser/NodeTraverser.php#L100-L162)), we can name it `NodeReplacement`:

```php
abstract public function mutate(Node $node): NodeReplacement;
```

This one also implies singular and plural.

* `SingleNodeReplacement implements NodeReplacement`
* `ArrayNodeReplacement implements NodeReplacement`
* ...